lib/deploy/stepFunctions/compileIamRole.js

```diff
@@ -642,27 +660,27 @@ function getS3ObjectPermissions(action, state) {
     return [
       {
         action: 's3:Get*',
-        resource: [
-          `arn:aws:s3:::${bucket}`,
-          `arn:aws:s3:::${bucket}/*`,
-        ],
+        resource: resolveS3BucketReferences(bucket, [
+          `arn:aws:s3:::\${bucket}`,
```

Comment:
I decided to keep the strings as template literals and just escape the `${bucket}` so it wasn't interpolated here, but in the `resolveS3BucketReferences()` function.

This is because other affected strings below uses other variables such as `${prefix}` and `${key}` and so they needed to remain template literals to have those interpolated correctly.

So I decided for consistency I would keep them all as template literals and uniformly escape the `${bucket}`.

Reply:
After review, I changed my mind and decided to use unescaped single quoted strings with a comment as I felt it was easier for the reader to understand why `bucket` wasn't being interpolated directly here.